crates/router/src/core/routing.rs

```diff
@@ -1182,19 +1183,20 @@ pub async fn toggle_dynamic_routing(
         id: profile_id.get_string_repr().to_owned(),
     })?;
 
-    let mut dynamic_routing_algo_ref: routing_types::DynamicRoutingAlgorithmRef = business_profile
-        .dynamic_routing_algorithm
-        .clone()
-        .map(|val| val.parse_value("DynamicRoutingAlgorithmRef"))
-        .transpose()
-        .change_context(errors::ApiErrorResponse::InternalServerError)
-        .attach_printable(
-            "unable to deserialize dynamic routing algorithm ref from business profile",
-        )?
-        .unwrap_or_default();
+    let mut success_based_dynamic_routing_algo_ref: routing_types::SuccessBasedDynamicRoutingAlgorithmRef =
+        business_profile
+            .dynamic_routing_algorithm
```

Comment:
In db both sr and elimination algo_id will be stored under this column itself right? So deserialise this field into global DynamicRoutingAlgorithmRef type which would have `success_rate_algo_id` and `elimination_algo_id`. Basically keep the same type as before but rename the `algorithm_id` to `success_rate_algorithm_id`

```
#[derive(Debug, Default, Clone, serde::Serialize, serde::Deserialize)]
pub struct DynamicRoutingAlgorithmRef {
    pub success_rate_algorithm_id: Option<common_utils::id_type::RoutingId>,
    pub timestamp: i64,
}
```

If you need independent timestamps for sr and elimination, tie it to its algo_id